src/Microsoft.VisualBasic/tests/VB/FileIOTests.vb

```diff
@@ -0,0 +1,963 @@
+            Using TestBase As New FileIOTests
+                Assert.Equal(FileSystem.GetName(TestBase.TestDirectory), IO.Path.GetFileName(TestBase.TestDirectory))
+            End Using
+        End Sub
+
+        <Fact>
+        Public Shared Sub GetParentPath_Path()
+            Using TestBase As New FileIOTests
+                Dim TestDirectory As String = TestBase.TestDirectory
+                Assert.Equal(FileSystem.GetParentPath(TestDirectory), IO.Path.GetDirectoryName(TestDirectory.TrimEnd(IO.Path.DirectorySeparatorChar, IO.Path.AltDirectorySeparatorChar)))
+            End Using
+        End Sub
+
+        <Fact>
+        Public Shared Sub GetTempFileName()
+            Dim TempFile As String = FileSystem.GetTempFileName
+            Assert.True(IO.File.Exists(TempFile))
+            Assert.True((New IO.FileInfo(TempFile)).Length = 0)
+            IO.File.Delete(TempFile)
+        End Sub
+
+        <Fact>
+        Public Shared Sub RenameDirectory_Directory_NewName()
+            Using TestBase As New FileIOTests
+                ''' <exception cref="IO.FileNotFoundException">If directory does not point to an existing directory.</exception>
+                Assert.Throws(Of IO.DirectoryNotFoundException)(Sub() FileSystem.RenameDirectory(IO.Path.Combine(TestBase.TestDirectory, "DoesNotExistDirectory"), "NewDirectory"))
+                Dim OrigDirectoryWithPath As String = IO.Path.Combine(TestBase.TestDirectory, "OriginalDirectory")
+                IO.Directory.CreateDirectory(OrigDirectoryWithPath)
+                ''' <exception cref="System.ArgumentException">If newName is Nothing or Empty String.</exception>
+                Assert.Throws(Of ArgumentNullException)(Sub() FileSystem.RenameDirectory(OrigDirectoryWithPath, ""))
+                Dim DirectoryNameWithPath As String = IO.Path.Combine(TestBase.TestDirectory, "DoesNotExist")
+                ''' <exception cref="System.ArgumentException">If contains path information.</exception>
+                Assert.Throws(Of ArgumentException)(Sub() FileSystem.RenameDirectory(OrigDirectoryWithPath, DirectoryNameWithPath))
+                FileSystem.RenameDirectory(OrigDirectoryWithPath, "NewFDirectory")
+                Dim NewFDirectoryPath As String = IO.Path.Combine(TestBase.TestDirectory, "NewFDirectory")
+                Assert.True(IO.Directory.Exists(NewFDirectoryPath))
+                Assert.False(IO.Directory.Exists(OrigDirectoryWithPath))
+                ''' <exception cref="IO.IOException">If directory points to a root directory or if there's an existing directory or an existing file with the same name.</exception>
+                IO.Directory.CreateDirectory(OrigDirectoryWithPath)
+                Assert.Throws(Of IO.IOException)(Sub() FileSystem.RenameDirectory(NewFDirectoryPath, "OriginalDirectory"))
+            End Using
+        End Sub
+
+        <Fact>
+        Public Sub CopyDirectory_SourceDirectoryName_DestinationDirectoryName()
```

Comment:
Why are some tests instance methods and others `Shared`? In these instance methods, there are two `FileIOTests` instances: `TestBase` and `Me`. Is that intentional?

Reply:
Some were shared because VB complains when they are not AND they don't use any instance items, this was a hopeless effort because CoreFx reference was written before Roslyn and VS had style guides so I gave up. I have removed all the shared in FIleIOTests. There should not be two FileIOTests instances but I also don't see two (Can't fine "Me")

Reply:
Are these tests from the reference source? If so, where?

The tests should either be instance methods (that use the implicit `Me` instance rather than a `TestBase` instance) or the tests should be `Shared` (with an explicit `TestBase` instance).

Reply:
Shared is removed from all test code except in the few places it is absolutely required. I don't understand the rest of the comment. No tests come directly from Reference Source, I was told they were not available and I couldn't find any.

Reply:
Sorry I wasn't clear. The tests should either be instance methods and should use fields from the implicit `Me` instance (for example, `Me.TestDirectory`) or the tests should be `Shared` and use an explicitly created instance. Since the tests create an explicit `TestBase As New FileIOTests()` (which seems reasonable given that the instance is explicitly disposed with a `Using` statement), there is no need for these methods to be instance methods.

In short, please add back `Shared` to each test method and to many of the helper methods.